plugins/providers/hyperv/scripts/set_enhanced_session_transport_type.ps1

```diff
@@ -17,8 +17,14 @@ try {
 }
 
 try {
-    Hyper-V\Set-VM -VM $VM -EnhancedSessionTransportType $Type
+    # HyperV 1.1 (Windows Server 2012R2) crashes on this call. Vagrantfiles before 2.2.10 do break without skipping this.
+    $present = Get-Command Hyper-V\Set-VM -ParameterName EnhancedSessionTransportType -ErrorAction SilentlyContinue
+    if($present) {
+        Hyper-V\Set-VM -VM $VM -EnhancedSessionTransportType $Type
+    }else{
+        Write-OutputMessage "This version of HyperV does not support EnhancedSessionTransportType, ignoring."
```

Comment:
It would be nice if this message was propagated up to the user, or at least output a debug line in the logs. For now if this runs on a machine without EnhancedSessionTransportType it will error because the output is expected to be [JSON](https://github.com/hashicorp/vagrant/blob/main/plugins/providers/hyperv/driver.rb#L74). You can output data from a powershell scripit by using `ConvertTo-Json` for example https://github.com/hashicorp/vagrant/blob/main/plugins/providers/hyperv/scripts/get_network_config.ps1#L56